pkg/controller.v1/pytorch/envvar.go

```diff
@@ -89,12 +112,26 @@ func setPodEnv(obj interface{}, podTemplateSpec *corev1.PodTemplateSpec, rtype,
 			// Set elastic related environment variables.
 			podTemplateSpec.Spec.Containers[i].Env = append(
 				podTemplateSpec.Spec.Containers[i].Env, envVars...)
+		} else {
+			podTemplateSpec.Spec.Containers[i].Env = append(
+				podTemplateSpec.Spec.Containers[i].Env, corev1.EnvVar{
+					Name:  EnvNnodes,
+					Value: strconv.Itoa(int(totalReplicas)),
+				})
```

Comment:
Yes it does, as I've mentioned above, nnodes is set in range format in elastic mode, e.g. nnodes=1:4, otherwise, nnodes is set by int, e.g. nnodes=2

Reply:
@kuizhiqing What env variable identifies nnodes for Elastic Mode ?
[As I can see here](https://github.com/kubeflow/training-operator/blob/731a12e2e6b759fe76509a23a29b77c6a01c69a1/pkg/controller.v1/pytorch/envvar.go#L115), for PyTorch Elastic we don't set `PET_NNODES`.

Reply:
`PET_NNODES` was set by `GetElasticEnvVarGenerator` in Elastic mode.

Reply:
@kuizhiqing @tenzen-y @johnugeorge Should we modify [the elastic policy env generator ](https://github.com/kubeflow/training-operator/blob/master/pkg/controller.v1/pytorch/elastic.go#L104-L108)to read from `.spec.nprocPerNode` instead of `.elasticPolicy.nprocPerNode` since we are going to deprecate the last one ?

Reply:
@andreyvelich Sorry, I missed your post.
I think adding validation to reject the request with both parameters would be good.

> > I think we just do not handle the inter-action between the two
> 
> +1
> 
> But can we add validation to restrict so that only either one can be set? This means that we don't allow creating PyTorchJob with both.spec.elasticPolicy.nProcPerNode and .spec.nprocPerNode .

https://github.com/kubeflow/training-operator/pull/1840#discussion_r1246233515

Reply:
Makes sense, validation should work.